backend/src/apiserver/server/visualization_server.go

```diff
@@ -88,6 +99,20 @@ func (s *VisualizationServer) generateVisualizationFromRequest(request *go_clien
 	return body, nil
 }
 
+func (s *VisualizationServer) getVisualizationServiceURL(request *go_client.CreateVisualizationRequest) (string, error) {
+	if common.IsMultiUserMode() {
+		if len(request.Namespace) == 0 {
+			return "", util.NewBadRequestError(
+				errors.New("Missing namespace in CreateVisualizationRequest."), "Namespace is request in multi-user mode.")
+		}
+		return fmt.Sprintf("http://%s.%s:%s",
+			common.GetStringConfig(visualizationServiceName),
```

Comment:
discussion: we've been using this pattern a lot, but I'm a little concerned with this.
Shall we move `common.GetXXXConfig` to class instance initiation phase instead and save the value to a class property for future usages?
Because `common.GetXXXConfig` is like global variables, they also affect how we write tests.

I'm not very sure about this feeling. Want to double check with you

Reply:
I don't have a strong preference. But in this particular case, I found that keeping the config read here is a little bit cleaner and more clear that they are relevant in multi-user mode only.

Reply:
SGTM